tests/unit/archive_validate_test.py

```diff
@@ -299,3 +306,160 @@ def test_run_summary_success(specs, expected_success):
+                    "encoding": "utf-8",
+                    "mediatype": "application/zip",
+                    "format": ".zip",
+                    "bytes": 21734463,
+                    "hash": "e9ece6bb8190e14d44834bf0a068ac5d",
+                },
+            ],
+            "created": "2023-11-30 20:51:43.255388",
+            "version": "1",
+        }
+    )
+)
+fake_new_datapackage_month_success = copy.deepcopy(fake_new_datapackage_quarter_success)
+# Make a successful monthly datapackage based on the quarterly one
+for resource in fake_new_datapackage_month_success.resources:
+    resource_months_min = pd.to_datetime(min(resource.parts["year_quarter"]))
+    resource_months_max = pd.to_datetime(
+        max(resource.parts["year_quarter"])
+    ) + relativedelta(months=3)
+    dd_idx = pd.date_range(start=resource_months_min, end=resource_months_max, freq="M")
+    resource.parts = {
+        "year_month": [pd.to_datetime(x).strftime("%Y-%m") for x in dd_idx]
+    }
+# Fails because it's missing 1997q1 - (last year, non-consecutive)
+fake_new_datapackage_quarter_fail1 = copy.deepcopy(fake_new_datapackage_quarter_success)
+fake_new_datapackage_quarter_fail1.resources[2].parts["year_quarter"] = ["1997q2"]
+# Fails because it's missing 1997-02 - (last year, non-consecutive)
+fake_new_datapackage_month_fail1 = copy.deepcopy(fake_new_datapackage_month_success)
+fake_new_datapackage_month_fail1.resources[2].parts["year_month"] = [
+    "1997-01",
+    "1997-03",
+]
+# Fails because it's missing 1996q4 - (middle year, non-complete)
+fake_new_datapackage_quarter_fail2 = copy.deepcopy(fake_new_datapackage_quarter_success)
+fake_new_datapackage_quarter_fail2.resources[1].parts["year_quarter"] = [
+    "1996q1",
+    "1996q2",
+    "1996q3",
+]
+# Fails because it's missing the rest of the months after 03.
+fake_new_datapackage_month_fail2 = copy.deepcopy(fake_new_datapackage_month_success)
+fake_new_datapackage_month_fail2.resources[1].parts["year_month"] = [
+    "1996-01",
+    "1996-02",
+    "1996-03",
+]
+# Test one year of data
+fake_new_datapackage_month_fail3 = copy.deepcopy(fake_new_datapackage_month_success)
+fake_new_datapackage_month_fail3.resources = [
+    fake_new_datapackage_month_fail3.resources[0]
+]
+
+
+@pytest.mark.parametrize(
+    "new_datapackage,success",
+    [
+        (fake_new_datapackage_quarter_success, True),
+        (fake_new_datapackage_month_success, True),
+        (fake_new_datapackage_quarter_fail1, False),
+        (fake_new_datapackage_month_fail1, False),
+        (fake_new_datapackage_quarter_fail2, False),
+        (fake_new_datapackage_month_fail2, False),
+        (fake_new_datapackage_month_fail3, True),
+    ],
+)
+def test_validate_data_continuity(new_datapackage, success):
+    """Test the dataset archiving valiation for epacems."""
+    validation = validate.validate_data_continuity(new_datapackage)
+    logger.info(validation)
+    if validation["success"] != success:
```

Comment:
can also just `assert validation['success'] == success`

Reply:
I feel like I always shy away from `assert` statements because I was once told they were unstable compared to `raise AssertionError`. But happy either way.